modules/cassandra/src/test/java/org/testcontainers/containers/CassandraContainerTest.java

```diff
@@ -0,0 +1,127 @@
+package org.testcontainers.containers;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import junit.framework.AssertionFailedError;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.CassandraQueryWaitStrategy;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Eugeny Karpov
+ */
+public class CassandraContainerTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(CassandraContainerTest.class);
+    private static final String TEST_STRING_IN_CONF = "._/test_string\\_.";
+
+    @Test
+    public void testSimple() throws Exception {
+        CassandraContainer cassandraContainer = (CassandraContainer) new CassandraContainer()
+                .withLogConsumer(new Slf4jLogConsumer(logger));
+        cassandraContainer.start();
+        try {
+            ResultSet resultSet = performQuery(cassandraContainer, "SELECT release_version FROM system.local");
+            assertTrue("Query was not applied", resultSet.wasApplied());
+            assertNotNull("Result set has no release_version", resultSet.one().getString(0));
+        } finally {
+            cassandraContainer.stop();
+        }
+    }
+
+    @Test
+    public void testSpecificVersion() throws Exception {
+        String cassandraVersion = "3.0.15";
+        CassandraContainer cassandraContainer = (CassandraContainer) new CassandraContainer("cassandra:" + cassandraVersion)
+                .withLogConsumer(new Slf4jLogConsumer(logger));
+        cassandraContainer.start();
+        try {
+            ResultSet resultSet = performQuery(cassandraContainer, "SELECT release_version FROM system.local");
+            assertTrue("Query was not applied", resultSet.wasApplied());
+            assertEquals("Cassandra has wrong version", cassandraVersion, resultSet.one().getString(0));
+        } finally {
+            cassandraContainer.stop();
+        }
+    }
+
+    @Test
+    public void testConfigurationOverride() throws Exception {
```

Comment:
this test does not test the override actually but rather demonstrates the issue with current implementation :D
If user provided a configuration and it's not found, we should fail instead of starting the container with our own, default config

Reply:
Good point! 👍  
Replaced by select of the changed cluster_name property from overridden configuration.

Reply:
Also added test that container won't start if user specified configuration directory with empty content.